lib/spack/spack/store.py

```diff
@@ -33,10 +34,76 @@
 import spack.config
 import spack.util.path
 import spack.database
-import spack.directory_layout as dir_layout
+import spack.directory_layout
+
 
 #: default installation root, relative to the Spack install path
-default_root = os.path.join(spack.paths.opt_path, 'spack')
+default_install_root = os.path.join(spack.paths.opt_path, 'spack')
+
+
+def parse_install_tree():
+    """Parse old and new formats and return relevant values.
```

Comment:
This docstring needs work.  "relevant values" does not tell me anything about what it returns, as a naive reader has no idea what is relevant.  This also doesn't tell me anything about the formats.

I'd expect there to be examples and descriptions of the formats here in the docstring, as well as description of the return type and what is in it.  Use the google docstring format we use elsewhere, and add `Arguments:` and `Returns:` sections.

For testability and clarity, this should just take a dict as a parameter (I guess it should expect the result of config.get("config")) instead of calling `config.get()`.  `_store` is the store-specific logic - let it call `config.get` and pass the result to `parse_install_tree`.  This thing should just process a dict.

Reply:
@tgamblin addressed